packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,305 @@
+}
+
+/**
+ * Takes a parameter property and returns the text of the equivalent class element.
+ */
+function getPropertyTextFromParam(param: ts.ParameterDeclaration): string {
+    const modifiers = param.modifiers ? param.modifiers.map((mod) => mod.getText()).join(' ') : '';
+    const typeAnnotations = param.type ? param.type.getText() : '';
+    return (
+        modifiers +
+        ' ' +
+        param.name.getText() +
+        (param.questionToken ? '?' : '') +
+        (!!typeAnnotations ? ': ' + typeAnnotations : '') +
+        ';'
+    );
+}
+
+/**
+ * Returns the super call expression index if it exists, otherwise -1.
+ */
+function getSuperCallIndex(construct: ts.ConstructorDeclaration): number {
+    const finalDirectiveIndex = getFinalDirectiveIndex(construct);
+    const stmts = construct.body!.statements;
+
+    /* It's possible that the body is comprised of directives only */
+    const firstNonDirectiveStmtIndex =
+        finalDirectiveIndex > -1
+            ? construct.body!.statements.length > finalDirectiveIndex + 1
+                ? finalDirectiveIndex + 1
+                : -1
+            : 0;
+
+    const stmt = firstNonDirectiveStmtIndex > -1 ? stmts[firstNonDirectiveStmtIndex] : undefined;
+    return stmt && isStatementSuperCall(stmt) ? firstNonDirectiveStmtIndex : -1;
+}
+
+/**
+ * Returns true when the prop is assigned any value other than the param identifier.
+ */
+function isNonParamAssignmentToProp(stmt: ts.ExpressionStatement, param: ts.ParameterDeclaration): boolean {
+    return (
+        isBinaryExpression(stmt.expression) &&
+        isPropertyAccessExpression(stmt.expression.left) &&
+        stmt.expression.left.expression.kind === ts.SyntaxKind.ThisKeyword &&
+        getPropertyName(stmt.expression.left.name) === param.name.getText() &&
+        (!isIdentifier(stmt.expression.right) ||
+            (isIdentifier(stmt.expression.right) && stmt.expression.right.text !== param.name.getText()))
+    );
+}
+
+function isPropertyMatchForParam(mem: ts.ClassElement, param: ts.ParameterDeclaration): boolean {
+    return (
+        isPropertyDeclaration(mem) &&
+        !mem.initializer &&
+        (!mem.questionToken ? !param.questionToken : !!param.questionToken) &&
+        getPropertyName(mem.name) === param.name.getText() &&
+        (!!mem.type ? !!param.type && param.type.getText() === mem.type.getText() : !param.type)
+    );
+}
+
+/**
+ * Checks if a statement assigns the parameter to a class member property.
+ */
+function isSimpleParamToPropAssignment(stmt: ts.Statement, param: ts.ParameterDeclaration): boolean {
+    return (
+        isExpressionStatement(stmt) &&
+        isBinaryExpression(stmt.expression) &&
+        isPropertyAccessExpression(stmt.expression.left) &&
+        stmt.expression.left.expression.kind === ts.SyntaxKind.ThisKeyword &&
+        isIdentifier(stmt.expression.right) &&
+        stmt.expression.left.name.getText() === param.name.getText() &&
```

Comment:
`getPropertyName`

Reply:
that not necessary, because `left.name` is always an Identifier. you can just access `left.name.text` directly